core/src/test/scala/unit/kafka/server/KafkaServerTest.scala

```diff
@@ -202,4 +202,16 @@ class KafkaServerTest extends QuorumTestHarness {
     TestUtils.createServer(kafkaConfig)
   }
 
+  @Test
+  def testGeneratedBrokerIdForbiddenDuringMigration(): Unit = {
+    val props = TestUtils.createBrokerConfig(-1, zkConnect)
+    props.put("reserved.broker.max.id", "2000")
```

Comment:
Does this test need to set `broker.id.generation.enable`? or is that implicit when `reserved.broker.max.id` is set

Reply:
I removed this, since the config is now enforced in `KafkaConfigTest`.